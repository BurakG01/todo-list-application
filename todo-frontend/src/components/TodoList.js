import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrash, faCheck } from "@fortawesome/free-solid-svg-icons";
import ListGroup from "react-bootstrap/ListGroup";
import "react-toastify/dist/ReactToastify.css";
import PropTypes from "prop-types";
export default class TodoList extends React.Component {
  static propTypes = {
    todos: PropTypes.array.isRequired,
    updateTodo: PropTypes.func,
    deleteTodo: PropTypes.func.isRequired,
  };
  render() {
  
    return (
      <ListGroup>
        { this.props.todos.map(({ _id, task, completed }, i) => (
          <ListGroup.Item
            className="todoList"
            style={{ outline: "none" }}
            key={i}
            variant={completed ? "dark" : ""}
            action
            onClick={(e) =>  this.props.updateTodo(e, _id)}
          >
            {task}

            <FontAwesomeIcon
              style={{ float: "right" }}
              onClick={(e) =>  this.props.deleteTodo(e, _id)}
              icon={faTrash}
            />
            {completed ? (
              <FontAwesomeIcon style={{ float: "right" }} icon={faCheck} />
            ) : null}
          </ListGroup.Item>
        ))}
      </ListGroup>
    );
  }
}
