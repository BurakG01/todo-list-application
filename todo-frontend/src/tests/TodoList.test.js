import TodoList from "../components/TodoList";

it("renders correctly", () => {
  const wrapper = shallow(
    <TodoList todos={[{_id:1,completed:false,task:"test"}]} deleteTodo={() => {}} updateTodo = {() => {}} />
  );

  expect(wrapper).toMatchSnapshot();
});

it("task correctly", () => {
  
    const wrapper = mount(
        <TodoList todos={[{_id:1,completed:false,task:"test"}]} deleteTodo={() => {}} updateTodo = {() => {}} />
    );
  
    const text = wrapper.find("ListGroupItem").text();
  
    expect(text).toEqual("test");
  });

  it("calls updateTodo", () => {
  
    const spy = sinon.spy();
  
    const wrapper = mount(
        <TodoList todos={[{_id:1,completed:false,task:"test"}]} updateTodo={spy} deleteTodo = {() => {}} />
    );
  
    wrapper
      .find("ListGroupItem")
      .first()
      .simulate("click");
  
    expect(spy.calledOnce).toBe(true);
  });

  it("calls deleteTodo", () => {
  
    const spy = sinon.spy();
  
    const wrapper = mount(
        <TodoList todos={[{_id:1,completed:false,task:"test"}]} updateTodo= {() => {}} deleteTodo ={spy} />
    );
  
    wrapper
      .find("FontAwesomeIcon")
      .first()
      .simulate("click");
  
    expect(spy.calledOnce).toBe(true);
  });