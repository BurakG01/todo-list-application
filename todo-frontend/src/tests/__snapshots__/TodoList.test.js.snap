// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ListGroup>
  <ListGroupItem
    action={true}
    active={false}
    className="todoList"
    disabled={false}
    key="0"
    onClick={[Function]}
    style={
      Object {
        "outline": "none",
      }
    }
    variant=""
  >
    test
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f1f8",
            "M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z",
          ],
          "iconName": "trash",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "float": "right",
        }
      }
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </ListGroupItem>
</ListGroup>
`;
